packages/grid/x-data-grid/src/hooks/features/editRows/useGridCellEditing.new.ts

```diff
@@ -290,18 +294,30 @@ export const useGridCellEditing = (
         : { ...row, [field]: value };
 
       if (processRowUpdate) {
-        Promise.resolve(processRowUpdate(rowUpdate, row)).then((finalRowUpdate) => {
-          apiRef.current.updateRows([finalRowUpdate]);
-          updateFocusedCellIfNeeded();
-          updateOrDeleteFieldState(id, field, null);
-        });
+        const handleError = (errorThrown: any) => {
+          if (onProcessRowUpdateError) {
+            onProcessRowUpdateError(errorThrown);
+          }
+        };
+
+        try {
+          Promise.resolve(processRowUpdate(rowUpdate, row))
+            .then((finalRowUpdate) => {
+              apiRef.current.updateRows([finalRowUpdate]);
+              updateFocusedCellIfNeeded();
+              updateOrDeleteFieldState(id, field, null);
+            })
+            .catch(handleError);
+        } catch (errorThrown) {
+          handleError(errorThrown);
+        }
```

Comment:
In terms of DX I'm wondering if the error should propagate when `onProcessRowUpdateError` is not set.

I imagine a dev trying to add serverinteraction which does not work due to an error in `processRowUpdate`, and they do not received it because it's catch by a function doing nothing

Reply:
I added a `console.error` when `onProcessRowUpdateError` is missing. If we propagate the error and it's thrown inside the promise, then the error boundary won't catch it. The developer has to look at the console.